plugins/redfish/fu-redfish-multipart-device.c

```diff
@@ -19,26 +19,72 @@ G_DEFINE_TYPE(FuRedfishMultipartDevice, fu_redfish_multipart_device, FU_TYPE_RED
 
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(curl_mime, curl_mime_free)
 
-static GString *
-fu_redfish_multipart_device_get_parameters(FuRedfishMultipartDevice *self)
+static void
+fu_redfish_multipart_device_get_parameters_generic(FuRedfishMultipartDevice *self,
+						   JsonBuilder *builder)
 {
-	g_autoptr(GString) str = g_string_new(NULL);
-	g_autoptr(JsonBuilder) builder = json_builder_new();
-	g_autoptr(JsonGenerator) json_generator = json_generator_new();
-	g_autoptr(JsonNode) json_root = NULL;
+	json_builder_begin_object(builder);
+	json_builder_set_member_name(builder, "Targets");
+	json_builder_begin_array(builder);
+	if (!fu_device_has_private_flag(FU_DEVICE(self), FU_REDFISH_DEVICE_FLAG_WILDCARD_TARGETS)) {
+		const gchar *logical_id = fu_device_get_logical_id(FU_DEVICE(self));
+		json_builder_add_string_value(builder, logical_id);
+	}
+	json_builder_end_array(builder);
+	json_builder_set_member_name(builder, "@Redfish.OperationApplyTime");
+	json_builder_add_string_value(builder, "Immediate");
+	json_builder_end_object(builder);
+}
 
-	/* create header */
+static void
+fu_redfish_multipart_device_get_parameters_smc(FuRedfishMultipartDevice *self, JsonBuilder *builder)
+{
+	/* https://supermicro.com/manuals/other/RedishRefGuide.pdf */
```

Comment:
My gut instinct here is that /redfish/v1/UpdateService/Actions/UpdateService.StartUpdate is sufficiently different that you want to create a `FuRedfishSupermicroDevice` device type and put all the differences there.